nlmod/util.py

```diff
@@ -113,265 +113,33 @@ def get_model_ds_empty(model_ds):
-                else:
-                    logger.info('cached data grid differs from current model')
-                    return False
-            if check_time:
-                if len(model_ds['time']) != len(model_ds2['time']):
-                    check_time = False
-                else:
-                    check_time = (model_ds['time'].data ==
-                                  model_ds2['time'].data).all()
-                if check_time:
-                    logger.info('cached data has same time discretisation as current model')
-                else:
-                    logger.info('cached data time discretisation differs from current model')
-                    return False
-            return True
-        except KeyError:
-            return False
-
-    elif model_ds.gridtype == 'unstructured':
-        try:
-            if check_grid:
-                check_cid = (model_ds['cid'] == model_ds2['cid']).all()
-                check_x = (model_ds['x'] == model_ds2['x']).all()
-                check_y = (model_ds['y'] == model_ds2['y']).all()
-                check_layer = (model_ds['layer'] == model_ds2['layer']).all()
-                # also check layer for dtype
-                check_layer = check_layer and (
-                    model_ds['layer'].dtype == model_ds2['layer'].dtype)
-                if (check_cid and check_x and check_y and check_layer):
-                    logger.info('cached data has same grid as current model')
-                else:
-                    logger.info('cached data grid differs from current model')
-                    return False
-            if check_time:
-                # check length time series
-                if model_ds.dims['time'] == model_ds2.dims['time']:
-                    # check times time series
-                    check_time = (model_ds['time'].data ==
-                                  model_ds2['time'].data).all()
-                else:
-                    check_time = False
-                if check_time:
-                    logger.info(
-                            'cached data has same time discretisation as current model')
-                else:
-                    logger.info(
-                            'cached data time discretisation differs from current model')
-                    return False
-            return True
-        except KeyError:
-            return False
-    else:
-        raise ValueError('no gridtype defined cannot compare model datasets')
-
-
-def get_cache_netcdf(use_cache, cachedir, cache_name, get_dataset_func,
-                     model_ds=None, check_grid=True,
-                     check_time=True, **get_kwargs):
-    """Create, read or modify cached netcdf files of a model dataset.
-
-    Steps:
 
-    1. Read cached dataset and merge this with the current model_ds if all
-    of the following conditions are satisfied:
+    return model_ds[list(model_ds.coords)]
```

Comment:
is this a copy, or is it still linked to the original data?

Reply:
Good question. I am not 100% sure but I think it is a copy. If I later add data variables to model_ds these are not added to the returned dataset from `get_model_ds_empty`. I also tried modifying the original coördinates of model_ds but could not do this. Therefore I think it is fine to get an empty model dataset using this code. To be sure we can also add `.copy(deep=True)` at the end. This is slower but not extremely slow, see timing below:

![image](https://user-images.githubusercontent.com/10562903/145586718-cdf4e6eb-fcc0-40b8-b1be-d877ceaff80c.png)

What do you think?

Reply:
Perhaps we can check by checking the memory id for the object, the dataarray and underlying array? `id(model_ds["x"])` etc?

But I'm also fine with the `.copy()` for now just to be sure.

Reply:
The id's are the same but they differ everytime I call the `id` function which is a bit confusing. I will use `copy()` for now